examples/react/navigation-blocking/src/main.tsx

```diff
@@ -80,6 +78,29 @@ function Editor1Component() {
         onChange={(e) => setValue(e.target.value)}
         className="border"
       />
+      {showConfirm && (
```

Comment:
I think the current setup for the "navigation blocking" example no longer adequately shows off the navigation blocking functionality, since it should demonstrate it using both `window.confirm` and a custom dialog.

We can change this after merge (if need be), but I'd rather go for a flow like this:

```
/
/using-window-confirm
/using-custom-ui
```

Where the `/using-window-confirm` and `/using-custom-ui` pages have a input field (like we have currently). When the fields have some value, it then requires user confirmation before allowing a navigation away.

Reply:
The example has two pages, I used the promise approach on a page and left the window.confirm in the other page.

Anyway it's indeed worth having two separate examples to keep one entirely with window.confirm and in the other we can showcase both approaches with the hook and with the component.

Reply:
IMO, having two separate examples mostly just creates more steps for people to have to go through. Ideally, I want to find out about navigation blocking, I either go to its guide or checkout its example. Introducing another decision in that flow is unnecessary.

Having it in a single example, the exact same content and logic, just separated by the two individual routes (with custom UI and without), in my mind makes the most sense. Plus, it also saves on waiting time for waiting for each Stackblitz example to load up.

Reply:
Makes sense! Would you like me to refactor it on this PR or we're keeping this as is and revisit it later on another PR?

Reply:
Probably for the time being have like `useState` toggle between `window.confirm` and the custom UI for simplicity's sake.

This can be made *prettier* afterwards.